workflow/controller/controller.go

```diff
@@ -148,7 +148,7 @@ func NewWorkflowController(restConfig *rest.Config, kubeclientset kubernetes.Int
 
 // RunTTLController runs the workflow TTL controller
 func (wfc *WorkflowController) runTTLController(ctx context.Context) {
-	ttlCtrl := ttlcontroller.NewController(wfc.wfclientset, wfc.wfInformer)
+	ttlCtrl := ttlcontroller.NewController(wfc.wfclientset, wfc.wfInformer, wfc.Config.WorkflowDefaults.Spec.GetTTLStrategy())
```

Comment:
Passing default TTLStrategy to TTL Controller

Reply:
this will not change if the config is edited - instead you should just do 👍 
`wfc.Config.WorkflowDefaults.Spec.GetTTLStrategy)`

Can `wfc.Config.WorkflowDefaults` be `nil`

Reply:
This is a not trivial fix, and I think you need to make it support reconfiguration at runtime of the to controller.

I suggest you have `configSupplied: func() Config`